### ISSUE TITLE
$I_MPI_WAIT_MODE is no longer supported by recent Intel MPI versions

### ISSUE THREAD

**Issue by boegel** [gjbex/worker#47]
worker defines `$I_MPI_WAITMODE` in the job script that it generates:

```
export I_MPI_WAIT_MODE=enable
```

This causes trouble with recent Intel MPI versions (tested with Intel MPI 2021 update 4 + 2019 update 9):

```
Abort(566799): Fatal error in PMPI_Init: Other MPI error, error stack:
MPIR_Init_thread(82)...:
MPIR_T_cvar_init(19376): Unable to parse environment variable (I_MPI_WAIT_MODE)
```

Apparently support for `$I_MPI_WAIT_MODE` was removed in Intel MPI recently, see also https://community.intel.com/t5/Intel-oneAPI-HPC-Toolkit/I-MPI-WAIT-MODE-replacement-in-Intel-MPI/m-p/1173203.

Is it strictly necessary to set `$I_MPI_WAIT_MODE`, or would it be OK to just strip this out?
Will `worker` still work correctly if it's not set (perhaps suboptimally)?

**Comment by gjbex** [comment]
We noticed similar issues.  It is okay to simply leave out I_MPI_WAIT_MODE, the consequence is that slaves which are no longer active will spin, but worker will work correctly.

The risk of load imbalance may increase as well when the -master flag is used, but that would mostly impact scenarios where worker is used inappropirately in the first place.

**Comment by gjbex** [comment]
The bugifix/conformance branch (1.6.14) removes I_MPI_WAIT_MODE and has some code clean up.

**Comment by stevenvdb** [comment]
I don't want to reopen the discussion on whether or not worker needs I_MPI_WAIT_MODE. I just want to mention that this variable is still supported (for sure in Intel(R) MPI Library, Version 2021.6, [click](https://www.intel.com/content/www/us/en/develop/documentation/mpi-developer-reference-linux/top/environment-variable-reference/other-environment-variables.html)), but now only supports 0 or 1 as values (no longer enable or disable).

**Comment by gjbex** [comment]
Indeed, I_MPI_WAIT_MODE was resurrected in 19 update 5, probably because its intended alternative didn't live up to expectations.
